src/main/java/com/mcmoddev/mmdcommunity/commoble/explodingchickens/ExplodingChicken.java

```diff
@@ -0,0 +1,177 @@
+package com.mcmoddev.mmdcommunity.commoble.explodingchickens;
+
+import java.lang.reflect.Field;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Supplier;
+
+import com.google.common.base.Suppliers;
+
+import net.minecraft.Util;
+import net.minecraft.network.syncher.EntityDataAccessor;
+import net.minecraft.network.syncher.EntityDataSerializers;
+import net.minecraft.network.syncher.SynchedEntityData;
+import net.minecraft.sounds.SoundEvents;
+import net.minecraft.util.Mth;
+import net.minecraft.world.effect.MobEffectInstance;
+import net.minecraft.world.entity.AreaEffectCloud;
+import net.minecraft.world.entity.EntityType;
+import net.minecraft.world.entity.ai.goal.Goal;
+import net.minecraft.world.entity.ai.goal.TemptGoal;
+import net.minecraft.world.entity.ai.goal.WrappedGoal;
+import net.minecraft.world.entity.animal.Chicken;
+import net.minecraft.world.entity.projectile.ThrownEgg;
+import net.minecraft.world.item.crafting.Ingredient;
+import net.minecraft.world.level.Explosion;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.gameevent.GameEvent;
+import net.minecraft.world.phys.Vec3;
+import net.minecraftforge.fml.util.ObfuscationReflectionHelper;
+
+public class ExplodingChicken extends Chicken
+{
+	private static final Supplier<Ingredient> FOOD = Util.make(() ->
+	{
+		Supplier<Field> field = Suppliers.memoize(() -> ObfuscationReflectionHelper.findField(Chicken.class, "f_28233_"));
```

Comment:
Break this into two parts ("f_" + "28233_") in case we ever update mappings on this repo.